exist-core/src/main/java/org/exist/xquery/value/Type.java

```diff
@@ -41,83 +42,105 @@
-    public final static int ANY_TYPE = 12;
-    public final static int ANY_SIMPLE_TYPE = 13;
-    public final static int UNTYPED = 14;
-    public final static int ATOMIC = 20;
-    public final static int UNTYPED_ATOMIC = 21;
-    public final static int STRING = 22;
-    public final static int BOOLEAN = 23;
-    public final static int QNAME = 24;
-    public final static int ANY_URI = 25;
-    public final static int BASE64_BINARY = 26;
-    public final static int HEX_BINARY = 27;
-    public final static int NOTATION = 28;
-    public final static int NUMBER = 30;
-    public final static int INTEGER = 31;
-    public final static int DECIMAL = 32;
-    public final static int FLOAT = 33;
-    public final static int DOUBLE = 34;
-    public final static int NON_POSITIVE_INTEGER = 35;
-    public final static int NEGATIVE_INTEGER = 36;
-    public final static int LONG = 37;
-    public final static int INT = 38;
-    public final static int SHORT = 39;
-    public final static int BYTE = 40;
-    public final static int NON_NEGATIVE_INTEGER = 41;
-    public final static int UNSIGNED_LONG = 42;
-    public final static int UNSIGNED_INT = 43;
-    public final static int UNSIGNED_SHORT = 44;
-    public final static int UNSIGNED_BYTE = 45;
-    public final static int POSITIVE_INTEGER = 46;
-    public final static int DATE_TIME = 50;
-    public final static int DATE = 51;
-    public final static int TIME = 52;
-    public final static int DURATION = 53;
-    public final static int YEAR_MONTH_DURATION = 54;
-    public final static int DAY_TIME_DURATION = 55;
-    public final static int GYEAR = 56;
-    public final static int GMONTH = 57;
-    public final static int GDAY = 58;
-    public final static int GYEARMONTH = 59;
-    public final static int GMONTHDAY = 71;
-    public final static int DATE_TIME_STAMP = 72;
-    public final static int TOKEN = 60;
-    public final static int NORMALIZED_STRING = 61;
-    public final static int LANGUAGE = 62;
-    public final static int NMTOKEN = 63;
-    public final static int NAME = 64;
-    public final static int NCNAME = 65;
-    public final static int ID = 66;
-    public final static int IDREF = 67;
-    public final static int ENTITY = 68;
-    public final static int JAVA_OBJECT = 100;
-    public final static int FUNCTION_REFERENCE = 101;
-    public final static int MAP = 102;
-    public final static int ARRAY = 103;
     private final static Logger LOG = LogManager.getLogger(Type.class);
 
-    private static int NO_SUCH_VALUE = -99;
-
-    private final static int[] superTypes = new int[512];
-    private final static Int2ObjectOpenHashMap<String[]> typeNames = new Int2ObjectOpenHashMap<>(64, Hash.FAST_LOAD_FACTOR);
-    private final static Object2IntOpenHashMap<String> typeCodes = new Object2IntOpenHashMap<>(100, Hash.FAST_LOAD_FACTOR);
+    // Tombstone value used for FastUtil Int maps
+    private static final int NO_SUCH_VALUE = -99;
```

Comment:
@adamretter why is this value `-99` in this case? I this a related to the FastUtil API?

Reply:
I just tested this. There is no technical reason to leave it at `-99` the tests all pass.

Reply:
Indeed, there is no technical reason why it is -1 or -99, technically it makes no difference. To be clear, I have never claimed otherwise. The reason it was changed was for human reasons:

1. When looking at it as an integer value it is much easier to spot.
2. When looking at it as an unsigned binary value the wrap causes it to look similar to other valid values, whereas -99 was an easier pattern to spot.

Both of these were factors as to why I changed it whilst developing and debugging this code. As it technically makes no difference, I suggest leaving it as -99 to assist the humans that will likely have to work on this code in future.

Reply:
Not convinced `-1` is very different from the other values as it is the only negative value.

Reply:
As the person who actually did the work on this and spent time debugging and fixing it when there were issues, I would hope that you would believe me when I am trying to tell you that I had problems spotting the `-1` amongst other values, and that switching it to `-99` made it easier to find the issues! As you did not do this work yourself, and did not have that experience personally, I find your comments perplexing!

As we agree that there is no technical advantage to changing the value, but that I have explained that there is a very real human advantage to leaving it as -99, please can you explain what your argument is for changing it to `-1`, i.e. What advantage do you see that causes you to be so insistent about this?

Reply:
lets make it `Integer.MIN_VALUE` then

Reply:
For what reason exactly?